components/code_panel.js

```
@@ -1,35 +1,87 @@
-            return new Date().getTime()
+        .flatMap(ev => {
+           var userCode = document.getElementById('user-code').value
+
+            // TODO: retrieve specs from the original JSON (store in context?)
+            var specs = `
+                            const expect = require('chai').expect
+                            const sum = require('subject')
+                            describe('sum', function () {
+                                it('should sum two numbers', function () {
+                                    expect(sum(1, 1)).to.equal(2)
+                                })
+                            })
+                        `
+
+            return Rx.Observable.fromPromise(tommy.run(userCode, specs)
+                .then(reporter => {
+                    var stats = reporter.stats
+
+                    var failed = stats.failures
+                    var passed = stats.passes
+                    var pending = stats.pending
+
+                    var numTests = stats.tests
+
+                    var result = { failed, passed, pending, numTests, status: 'pending' }
+
+                    if (failed > 0) {
+                        result.status = 'failed'
+                    }
+
+                    if (passed == numTests) {
+                        result.status = 'passed'
+                    }
+
+                    return result
+                })
+                .catch(err => {
+                    return { error: `Execution error: ${err}}` }
+                }))
         })
 
     return context
         .map(ctx => ctx.timestamp)
         .concat(buttonClicks$)
 }
 
-function view(timestamp$) {
-    return timestamp$.map(ts => {
-        return (
-            <div id="code">
-                <header>
-                    <div className="container-header">
-                        Your code
+function view(testResults$) {
+    return testResults$
+        .map(res => {
+            var resultsStatusClass = res.status
```

and instead of  `hidden` for status, you could use default status value like `status-unknown` and hide unknown status in css